feat(replay): query IP for trace connected errors for replay summary (v2)

michellewzhang opened the pull request:
redo of #97737

reverted the old pr as the trace id query was too slow -- it did 2 snuba queries per trace id (bad). this pr instead makes it  2 total queries for all trace ids

MERGE WED

michellewzhang committed:
Revert "Revert "feat(replay): query IP for trace connected errors for replay summary (#97737)""

This reverts commit c40e63c3935ab139ca357740bf1a1e14c2d240ab.

michellewzhang committed:
:recycle: batch

aliu39 commented on src/sentry/replays/lib/summarize.py:
```diff
@@ -89,65 +89,60 @@ def fetch_trace_connected_errors(
+        )
 
-        if not queries:
+        if not (error_query or issue_query):
```
I don't think we need this since we check `if error_query` and `if issue_query` below

michellewzhang committed:
:pencil2: rename

aliu39 reviewed (approved):
🤞

aliu39 commented:
Branching off this and adding better tracing/logging to debug

aliu39 commented on src/sentry/replays/lib/summarize.py:
```diff
@@ -90,63 +89,112 @@ def fetch_trace_connected_errors(
+            ],
+            query=trace_ids_query,
+            snuba_params=snuba_params,
+            orderby=["event_id"],
```
maybe lets order by most recent, `"-timestamp"`

michellewzhang committed:
:recycle: order by time

aliu39 committed:
Merge branch 'master' of github.com:getsentry/sentry into mz/fix-ip-query

aliu39 committed:
Fix merge

aliu39 committed:
Fix dup handling

aliu39 committed:
Ref

aliu39 commented on src/sentry/replays/usecases/summarize.py:
```diff
@@ -313,8 +313,7 @@ def generate_summary_logs(
         if error["category"] == "error":
             yield generate_error_log_message(error)
-        elif error["category"] == "feedback":
-            seen_feedback_ids.add(error["id"])
```
The set isn't used in this loop or after it. We needed to check if the id was in the list, since it could've been added during the breadcrumb parsing

aliu39 commented on src/sentry/replays/usecases/summarize.py:
```diff
@@ -90,63 +89,112 @@ def fetch_trace_connected_errors(
+        # Query for issuePlatform dataset
+        issue_query_results = query_trace_connected_events(
+            dataset_label="issuePlatform",
+            selected_columns=[
+                "event_id",
+                "title",
+                "subtitle",
+                "timestamp",
+                "occurrence_type_id",
```
cool added to linear

aliu39 replied:
ah ok!

aliu39 committed:
ref(replay): add tracing and metrics to summary endpoint (#98494)

Wrap the methods of this endpoint with a custom transaction. Sampling
rates will be options-controlled. We need a high POST sample rate while
we develop this feature, to debug slow loading times.

I believe options can be segmented by user email, or something like
is_employee - will experiment
Merge with https://github.com/getsentry/sentry/pull/98479

aliu39 committed:
Merge branch 'master' into mz/fix-ip-query

michellewzhang commented on src/sentry/replays/endpoints/project_replay_summary.py:
```diff
@@ -133,91 +140,124 @@ def has_replay_summary_access(self, project: Project, request: Request) -> bool:
-        filter_params = self.get_filter_params(request, project)
-        num_segments = request.data.get("num_segments", 0)
-        temperature = request.data.get("temperature", None)
+            # Fetch errors linked in breadcrumbs.
```
```suggestion
            # Fetch errors from breadcrumbs.
```

michellewzhang committed:
Update src/sentry/replays/endpoints/project_replay_summary.py

aliu39 committed:
Dont check dups in fetch_trace_connected and expect data key

aliu39 committed:
Rewrite seen set in generate_summary_logs

aliu39 committed:
Handle dup error (direct and trace-conn)

aliu39 commented on src/sentry/replays/endpoints/project_replay_summary.py:
```diff
@@ -212,7 +212,7 @@ def post(self, request: Request, project: Project, replay_id: str) -> Response:
             trace_ids = processed_response[0].get("trace_ids", []) if processed_response else []
 
-            # Fetch errors linked in breadcrumbs.
+            # Fetch errors from breadcrumb events.
```
I realized these aren't actually linked in the breadcrumb events..? Not sure, I just call them direct-connected, direct-linked

michellewzhang replied:
yeah just direct

michellewzhang commented on src/sentry/replays/usecases/summarize.py:
```diff
@@ -135,69 +135,50 @@ def fetch_trace_connected_errors(
+                )
 
         # Process issuePlatform query results
-        if issue_query_results and "data" in issue_query_results:
```
is there a chance the query results don't exist?

aliu39 replied:
I think that's only when snuba errors, and we should raise rather than silently skip
